src/DynamoCoreWpf/Views/HomePage/HomePage.xaml.cs

```diff
@@ -0,0 +1,572 @@
+        {
+            string htmlString = string.Empty;   
+            string jsonString = string.Empty;
+
+            // When executing Dynamo as Sandbox or inside any host like Revit, FormIt, Civil3D the WebView2 cache folder will be located in the AppData folder
+            var userDataDir = new DirectoryInfo(GetUserDirectory());
+            PathHelper.CreateFolderIfNotExist(userDataDir.ToString());
+            var webBrowserUserDataFolder = userDataDir.Exists ? userDataDir : null;
+
+            dynWebView.CreationProperties = new CoreWebView2CreationProperties
+            {
+                UserDataFolder = webBrowserUserDataFolder.FullName
+            };
+
+            //ContentRendered ensures that the webview2 component is visible.
+            await dynWebView.Initialize();
+            // Context menu disabled
+            this.dynWebView.CoreWebView2.Settings.AreDefaultContextMenusEnabled = false;
+            // Zoom control disabled
+            this.dynWebView.CoreWebView2.Settings.IsZoomControlEnabled = false;
+            this.dynWebView.CoreWebView2.Settings.AreDevToolsEnabled = true;
+
+            var assembly = Assembly.GetExecutingAssembly();
+
+            using (Stream stream = assembly.GetManifestResourceStream(htmlEmbeddedFile))
+            using (StreamReader reader = new StreamReader(stream))
+            {
+                htmlString = reader.ReadToEnd();
+            }
+
+            using (Stream stream = assembly.GetManifestResourceStream(jsEmbeddedFile))
+            using (StreamReader reader = new StreamReader(stream))
+            {
+                var jsString = reader.ReadToEnd();  
+                jsonString = jsString;
+            }
+
+            // Copy the embedded resource to a temporary folder
+            using (var stream = Assembly.GetExecutingAssembly().GetManifestResourceStream(fontStylePath))
+            {
+                if (stream != null)
+                {
+                    var fontData = new byte[stream.Length];
+                    stream.Read(fontData, 0, fontData.Length);
+
+                    // Create the temporary folder if it doesn't exist
+                    Directory.CreateDirectory(virtualFolderPath);
+
+                    // Write the font file to the temporary folder
+                    fontFilePath = Path.Combine(virtualFolderPath, "ArtifaktElement-Regular.woff");
+                    File.WriteAllBytes(fontFilePath, fontData);
+                }
+            }
+
+            // Set up virtual host name to folder mapping
+            dynWebView.CoreWebView2.SetVirtualHostNameToFolderMapping(virtualFolderName, virtualFolderPath, CoreWebView2HostResourceAccessKind.DenyCors);
+
+            htmlString = htmlString.Replace("mainJs", jsonString);
+
+            try
+            {
+                dynWebView.NavigateToString(htmlString);
+            }
+            catch (Exception ex)
+            {
+                Debug.WriteLine(ex.Message);
+            }
```

Comment:
I feel like so far, these code are repeated in other places. Is there any chance we can move them to a shared utility class, like DynamoUtilities project?

Reply:
I did my best to factor out the functions I thought were repetitive. I placed them provisionally inside the `PathHelper` class, as I found that most relevant. Let me know if the change is not what you expected and I will update as needed, @QilongTang